### ISSUE TITLE
MDX parsing error for components that are not self-closing (fixed, but not yet released)

### ISSUE THREAD

**Issue by robinmetral** [prettier/prettier#7176]
**Prettier 1.19.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBadAdWBLGAbOJAAgAkIBbOLdVLLbcgBwgCcYjgiAZbKAayIBfIgDMWFIhhABzAIYwAzgCMAnlLpQAxEQAKs6XCIxcBDQGUKhxrJb7bjABYaAPD35GIAXikB6BZYB3BzgWOFQ4PAU4HykAPgBxCCJ-SiCQwwio5x83PlisIiIQABoQCEZjaAVkUBtxAL1QqGqUWQA3CGwAExKQJVswPjgYM2swXmlkGBYAVzhShxhyPAB1B1w4BTG4MwQFXGw23BVkcAVq0t4oth1baXJZZBFZSPmQACsFAA8AIQGhkayShuOBPF5RUqfL5mCYEACKMwg8DBr1K1hY11O5C6X16jBYvBgK26MAcyAAHAAGNHiKIreynfGbEJtUGlACOiPgt3KLRAsgUqCgcDgXVFvVCnOwoVu+geKIhICi5GwU1mb32UGk8K5oKQz1RIBgsiUxK6pOQACZStNZNg8BMAMIUeUoKDQNkgGZRAAqJpaBsVbTmAEkoGLYGYwASKgBBcNmGAqAgKuCCQRAA)
```sh
--parser mdx
```

**Input:**
```mdx
---
title: Home
---

import { Link } from "gatsby"

# Page title

Some paragraph

<Link to="/somewhere-else/">Go somewhere else</Link>
  
```

**Output:**
```mdx
SyntaxError: Void elements do not have end tags "Link" (1:46)
> 1 | ---
    |    ^
  2 | title: Home
  3 | ---
  4 | 
```

**Expected behavior:**

This should not throw an error as it is valid MDX.

The following does not throw an error ([Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBadAdWBLGAbOJAAgAkIBbOLdVLLbcgBwgCcYjgiAZbKAayIBfIgDMWFIhhABzAIYwAzgCMAnlLpQAxEQAKs6XCIxcBDQGUKhxrJb7bjABYaAPD35GIAXikB6BZYB3BzgWOFQ4PAU4HykiPFklCO8QAHEIIn9KIJDDCKjYnwA+IiIQABoQCEZjaAVkUBtxAL1QqDqUWQA3CGwAE3KQJVswPjgYM2swXmlkGBYAVzgKhxhyPAB1B1w4BUm4MwQFXGxO3BVkcAU6it4oth1baXJZZBFZSKWQACsFAA8AIWGo3Gskobjgr3eUQqP1+ZmmBAAivMIPBIR8KtYWHcLuRer8BowWLwYOs+jAHMgABwABkx4ii63sFyJOxCnQhFQAjij4A8qu0QLIFKgoHA4L0JQNQjzsKEHvpnujoSAouRsLMFp8jlBpEjeRCkG8MSAYAkyb0KcgAEwVOaybB4aYAYQoSpQUGgnJA8yiABUEu1jSrOosAJJQSWwMxgYnVACCkbMMBUBGVcEEgiAA)):

```mdx
<Link to="/somewhere-else/" label="Go somewhere else" />
```

It seems that the error is coming from the component not being self-closing.

**Comment by kachkaev** [comment]
Indeed there is something strange happening with MDX in Prettier 1.19.1

✅ [`<Linc to="x">y</Linc>`](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAMgSymABDCAXgB0QAPUgPgE9UB6LHSkAGhAgAcZNoBnZUAEMATsIgB3AAoiE-FIIBuETABNWIAEbDBYANZwYAZQ47sAc2QxhAVzhsAFjAC2AGwDq9zPF4mwcQ7JemApe1MjgvPxs2LxwwjCS2mZOgsgAZoIusWwAVrxkAELaegaGgk5wjHDpmdkgeWSG5i5wAIrWEPA1WXYgJsKxwuFOKmTqHMLYMG6qMPbIABwADGwTELFu2hzhE3CDCtVsAI4d8ImcciCCvAC0UHBwKo-qwnAnmK+JgsmpSBk9bFiTkwlhsvV4zTap2qf1qvRggg0MxUc2QACY2FZBJgXOYAMIQJwpcJQaCHEDWWIAFURcn+dQUtgAklAnrBDGBJlwAIKswwwagtbqxAC+IqAA)  
❌ [`<Link to="x">y</Link>`](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAMgSygawAQwQC8AOiAB5kB8AnqgPRa5UgA0IEADjJtAM7JQAQwBOIiAHcACqIQCUQgG4RMAEzYgARiKFgccGAGVOu7AHNkMEQFc47ABYwAtgBsA6vczw+JsHENyXpiKXjTI4HwC7Nh8cCIwUjpmTkLIAGZCLrHsAFZ85ABCOnoGhkJOcExw6ZnZIHnkhuYucACK1hDwNVl2ICYisSLhTqrkGpwi2DBuajD2yAAcAAzsExCxbjqc4RNwg4rV7ACOHfCJXPIgQnwAtFBwcKqPGiJwJ5iviULJqUgZPexYk5MJYbL0+M02qdqn9ar0YEJNDNVHNkAAmdhWISYFzmADCECcKXCUGghxA1liABVEfJ-nVFLYAJJQJ6wQxgSbcACCrMMMBoLW6sQAviKgA)

✅ [`<div foo="bar">abc</div>`](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAJgSwG4AIBmEEAvADogBGAhgE7kB8VFYqA9FtvSADQgQAOMTNADOyULRoQA7gAVaCMSirYImdD0o0qYANZwYAZX47MUAObIYNAK5xeACxgBbADYB1B5ngiTYOIaK3jjeAJ7I4CJivGYicDQwstrmzlTI+FSucbwAViIAHgBC2noGhlTOcAAyZnDpmdkgefmGZuaucACKNhDw9Vn2ICY0cTQRzuj5mvw0ZjDu6jAOyAAcAAy8MxBx7tr8ETNwo9h1vACOPfBJAkogVCIAtFBwcOivmjRwF5ifSVQpaSQGQGvDizkwVlsgxEbQ63V6dSBDUGMCYC3QS2QACZeNYqJhXG0AMIQZypCJQaCnEA2OIAFSYSmBjWwdgAklA3rBDGBZoIAIKcwwwUIdfpxAC+EqAA)  
❌ [`<div foo={{ bar: true }}>abc</div>`](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAJgSwG4AIBmEEAvMMLgEYCGATkrjDQK5y4C+bAfFRWKgPRZsnEABoQEAA4xM0AM7JQtGhADuABVoIFKKtgiZ0YkBRpUwAazgwAypPOYoAc2SMW4gBYwAtgBsA6h6Y8HL2YHA22sE4wQCeyOByCuKOcnA0MOpmTt5UyPhUvmniAFZyAB4AQmaW1jZU3nAAMo5w+YXFIGXlNo5OvnAAikwQ8O1FcOL2NGk0Cd7o5caSNI4w-oYwHsgAHAAMUypp-maSCStws9ht4gCOI-BZUjogVHIAtFBwcOg-xjRwe6YAFZKg5PJIAoTcRpbyYVzMSYgOR9AbDUZtSEdJEwHgbdBbZAAJnEjComF8fQAwhBvLkElBoDcQEw0gAVHg6KGdbAsACSUF+sBsYFW0gAgoKbDBYgNxmkOEA)

✅[`<></>`](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAfKg9OkAaECABxgEtoBnZUAQwCc6IB3ABXoSpRoDcJSATfCABGdGmADWcGAGUi40lADmyGHQCucAgAsYAWwA2AdW2l4FeWDgyOZ0tzMBPZOApUCiinDowWYpXo0yABmNAZeBABWFAAeAEJiktIyNHpwADKKcCFhESDRMTKKSgZwAIrqEPA54Vog8nRedC56-DFCRHSKMEYCMNrIABwADASdEF5GYkQunXBN3NkEAI6V8H7EnCA0FAC0UHBw-EdCdHCrpGd+NAFBSKG1BF56pKoadRTFpRVV2fe5dRgNGEvX4-WQACYCGoaKQDMUAMIQPSBFxQaBLEDqLwAFWBnAeeW4mgAklBjrAZGAuiQAILkmQwRylGpeAC+bKAA)  
✅ [`<Playground></Playground>`](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeACgGwIYE8DmAThAK5QAmAfKgPRZ5GkWUgA0IEADjAJbQDOyUNkLEA7uhEJBKbADcIPcmxAAjQtjABrODADKnTTyj5kMQiTjsAFjAC2mAOrWe8fobBw901zzmvcZHB+QXZjfjhCGHQNfDtsZAAzbEwI9gArfgAPACENbV09bDs4ABljOCSUtJBMrL1jfEw4AEUSCHgq1KsQQ0IIwiC7ciyVTkJjGEclGGtkAA4ABnZxiAjHDU4g8bgBuUr2AEd2+BiuGRBsfgBaKDg4cgeVQjhjnheY7DiEpGTu9gidh4ZgsPX4jWabQ6lV+1R6MGwqmm5FmyAATOxzNgeJhGgBhCB2eJBKDQA4gEgRAAqiJkfxqcksAEkKAh9GAJtwAIIUPQwXDNLoRAC+wqAA)  
❌ [`<Playground><></></Playground>`](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeACgGwIYE8DmAThAK5QAmAfKtQPR1Z5GkWUgA0IEADjAJbQAzslDZCxAO7oxCYSmwA3CH3IcQAI0LYwAazgwAyt218o+ZDEIk4nABYwAtpgDqtvvEHGwcA7Pd8Fd1xkcEFhTlNBOEIYdC18B2xkADNsTCjOACtBAA8AIS1dfQNsBzgAGVM4FLSMkGycg1N8TDgARRIIeBr0mxBjQijCEIdyHLVuQlMYZxUYW2QADgAGTkmIKOctbhDJuCGFas4AR074OJ45EGxBAFooODhyJ7VCOFO+N7jsBKSkVN6nCiDj4FisfUEzVaHS61X+tT6MGw6lm5HmyAATJxLNg+JhmgBhCAORIhKDQI4gEhRAAqyLkALqCmsAEkKAhDGAprwAIIUAwwXCtHpRAC+oqAA)

This seems to be a new behaviour in 1.19 – the above examples did work in 1.18.x. Sadly, we are blocked by these oddities in MDX and have to stick with 1.18 for now. Prettier 1.19 is required to start using TypeScript 3.7 awesome features, so if anyone has ideas, please help 🙏

**Comment by kachkaev** [comment]
The above issues could have been fixed by https://github.com/prettier/prettier/pull/6949. Playground for that PR:

✅ [`<Link to="x">y</Link>`](https://deploy-preview-6949--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAeAMgSygawAQwQC8AOiAB5kB8AnqgPRa5UgA0IEADjJtAM7JQAQwBOIiAHcACqIQCUQgG4RMAEzYgARiKFgccGAGVOu7AHNkMEQFc47ABYwAtgBsA6vczw+JsHENyXpiKXjTI4HwC7Nh8cCIwUjpmTkLIAGZCLrHsAFZ85ABCOnoGhkJOcExw6ZnZIHnkhuYucACK1hDwNVl2ICYisSLhTqrkGpwi2DBuajD2yAAcAAzsExCxbjqc4RNwg4rV7ACOHfCJXPIgQnwAtFBwcKqPGiJwJ5iviULJqUgZPexYk5MJYbL0+M02qdqn9ar0YEJNDNVHNkAAmdhWISYFzmADCECcKXCUGghxA1liABVEfJ-nVFLYAJJQJ6wQxgSbcACCrMMMBoLW6sQAviKgA)  
✅ [`<div foo={{ bar: true }}>abc</div>`](https://deploy-preview-6949--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAeAJgSwG4AIBmEEAvMMLgEYCGATkrjDQK5y4C+bAfFRWKgPRZsnEABoQEAA4xM0AM7JQtGhADuABVoIFKKtgiZ0YkBRpUwAazgwAypPOYoAc2SMW4gBYwAtgBsA6h6Y8HL2YHA22sE4wQCeyOByCuKOcnA0MOpmTt5UyPhUvmniAFZyAB4AQmaW1jZU3nAAMo5w+YXFIGXlNo5OvnAAikwQ8O1FcOL2NGk0Cd7o5caSNI4w-oYwHsgAHAAMUypp-maSCStws9ht4gCOI-BZUjogVHIAtFBwcOg-xjRwe6YAFZKg5PJIAoTcRpbyYVzMSYgOR9AbDUZtSEdJEwHgbdBbZAAJnEjComF8fQAwhBvLkElBoDcQEw0gAVHg6KGdbAsACSUF+sBsYFW0gAgoKbDBYgNxmkOEA)  
✅ [`<Playground><></></Playground>`](https://deploy-preview-6949--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAeACgGwIYE8DmAThAK5QAmAfKtQPR1Z5GkWUgA0IEADjAJbQAzslDZCxAO7oxCYSmwA3CH3IcQAI0LYwAazgwAyt218o+ZDEIk4nABYwAtpgDqtvvEHGwcA7Pd8Fd1xkcEFhTlNBOEIYdC18B2xkADNsTCjOACtBAA8AIS1dfQNsBzgAGVM4FLSMkGycg1N8TDgARRIIeBr0mxBjQijCEIdyHLVuQlMYZxUYW2QADgAGTkmIKOctbhDJuCGFas4AR074OJ45EGxBAFooODhyJ7VCOFO+N7jsBKSkVN6nCiDj4FisfUEzVaHS61X+tT6MGw6lm5HmyAATJxLNg+JhmgBhCAORIhKDQI4gEhRAAqyLkALqCmsAEkKAhDGAprwAIIUAwwXCtHpRAC+oqAA)

Looking forward for 1.19.2 / 1.20.0 / 2.0.0  👀 🙏

**Comment by thorn0** [comment]
Duplicate of #6943

**Comment by thorn0** [comment]
Yes, fixed, but not yet released. Let's keep this issue open for a while, so that people can see it.

**Comment by JounQin** [comment]
Sorry for my fault, it reminds me that any changes of such OSS effect a lot of users.
Cheers.

**Comment by JDansercoer** [comment]
@kachkaev Not sure if this is a worthy addition to the discussion, but downgrading to `1.18.2` does not solve it. So maybe there's another underlying problem?

```
["INFO" - 12:38:24 PM] Formatting /xx/src/pages/index.mdx
["INFO" - 12:38:24 PM] Loaded module 'prettier@1.18.2' from '/xx/node_modules/prettier/index.js'
["INFO" - 12:38:24 PM] Using ignore file (if present) at /xx/.prettierignore
["INFO" - 12:38:24 PM] File Info:
{
  "ignored": false,
  "inferredParser": "mdx"
}
["INFO" - 12:38:24 PM] Detected local configuration (i.e. .prettierrc or .editorconfig), VS Code configuration will not be used
["INFO" - 12:38:24 PM] Prettier Options:
{
  "filepath": "xx/index.mdx",
  "parser": "mdx",
  "singleQuote": true,
  "trailingComma": "es5",
  "proseWrap": "always"
}
["ERROR" - 12:38:24 PM] Error formatting document.
Void elements do not have end tags "Link" (1:45)
```

Also using the same code example as said in the OP.

**Comment by thorn0** [comment]
@JDansercoer Try installing Prettier `master` from GitHub (`"prettier": "prettier/prettier"` instead of `"prettier": <version number>` in your `package.json`).

**Comment by JDansercoer** [comment]
@thorn0 `"prettier/prettier"` seems to work, thanks! Any idea when it will be released?

Also, maybe little bit off-topic, but how does the resolving to `"prettier/prettier"` work? Can't seem to find documentation on that. I only remember every including a direct link to Github?

**Comment by thorn0** [comment]
@JDansercoer Hopefully, Prettier 2.0 will be out in January. Follow #6888 for details. As for installing from GitHub, see https://docs.npmjs.com/cli/install.

**Comment by detj** [comment]
Is this error related to this issue?

```jsx
<div style={{width: '100px'}}></div>
```

[Playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAJgSwG4AIDOMAngDZwC8wwA7pujABZK4DkAjAAwcAOAHiwF8BAPlQB6LNmEAdKCAA0ICNxiZo+ZKACGAJx0RqABV0INKLdgh0FIAEY6tYANZwYAZW6PMUAObIYOgCucIoMMAC2JADqDJjw+J5gcG6mcThxRMjg+BqK3vhwOjCGDj7hWsgAZlokBYoAVvi8AEIOzq5uWuFwADLecFU1dSCNvG7ePmQAioEQ8IO1ISCeOgU6WeHovDbcOt4wUXSMyAAcHIq7EAVRDtxZu3Br2AOKAI6z8CXKZiBa+AC0UDgcHQIJsOjg70wEJKWjKFSQ1UWigK4Uw-iCS3wE2mHwGiKGSxgWlsh3oDGQACZFAEtJgSBMAMIQcLlLJQaAvECBAoAFRJZiRw2wwQAklBQbA3GA9ioAIIStzEMgLApCIA)

**Comment by fisker** [comment]
@detj 

https://github.com/prettier/prettier/pull/6773

[Playground](https://deploy-preview-6773--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAeAJgSwG4AIDOMAngDZwC8wwA7pujABZK4DkAjAAwcAOAHiwF8BAPlQB6LNmEgANCAjcYmaPmSgAhgCdNEagAUtCVSnUlq6oqrkAjTerABrODADK3e5igBzZDE0BXODkGGABbEgB1Bkx4fHcwOBcjGJwYomRwfCsQT3w4TRg9Oy9Q9WQAM1M8uQArfF4AITtHZxd1ULgAGU84CqqgkDreF08vMgBFfwh4PpJqkHdNPM0M0PReWQXNTxgIukZkAA4OOW4dPIi7bgyzuGXsXrkARyn4IoVjEHV8AFooODg6EBm00cBemFBRXUJTKSEqcwGeVCmF8AURowmr16cP6chg6mse3oDGQACY8XZMCRRgBhCChUoZO4AVk2-jyABUCcZ4fNsIEAJJQIGwFxgbaKACCwpcxDIszyQiAA)

**Comment by robinmetral** [comment]
@thorn0 Isn't it misleading to close this when it hasn't been fixed? A better practice IMHO would instead be to [reference it](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR that will solve it (#6773 as far as I understand).

**Comment by thorn0** [comment]
It has been fixed.

**Comment by robinmetral** [comment]
Not in the `latest`, [playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBadAdWBLGAbOJAAgAkIBbOLdVLLbcgBwgCcYjgiAZbKAayIBfIgDMWFIhhABzAIYwAzgCMAnlLpQAxEQAKs6XCIxcBDQGUKhxrJb7bjABYaAPD35GIAXikB6BZYB3BzgWOFQ4PAU4HykAPgBxCCJ-SiCQwwio5x83PliQABoQCEZjaAVkUBtxAL1QqAqUWQA3CGwAE0KQJVswPjgYM2swXmlkGBYAVzgihxhyPAB1B1w4BWG4MwQFXGxm3BVkcAUKot4oth1baXJZZBFZSJmQACsFAA8AIV7+wdlKNxwe6PKJFN7vMyjAgARUmEHgwKeRWsLAuRyUsiUES6jBYvBgiw6MAcyAAHAAGZHiKKLexHXFrELNIFFACOcPgVxKjRAsgUqCgcDg7WFXVC7OwoSu+luiNBICi5Gw4ymzx2UGkMI5QKQDyRIBgmMJ7WJyAATEUJrJsHhRgBhCiylBQaAskCTKIAFUxjT18ua0wAklARbAzGA8aUAIIhswwFQEOVwQSCIA)

**Comment by millette** [comment]
@robinmetral Hello Robin :-)

It's fixed in master, you can install (if you like bleeding edge, or just to test it):

```sh
npm install prettier/prettier
```

After, you should see in package.json:

```json
{
  "dependencies": { 
    "prettier": "github:prettier/prettier"
  }
}
```

Amongst others.

It works for me ™

**Comment by robinmetral** [comment]
Thanks for the info Robin @millette 🙂 

Can you point me to which PR fixed it? Another follow-up question would be why a bugfix merged to master doesn't trigger a release 😉

**Comment by fisker** [comment]
I think https://github.com/prettier/prettier/pull/6949 is the one
